applications/DEMApplication/custom_processes/apply_forces_and_moments_process.h

```diff
@@ -0,0 +1,243 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main author:     Joaquín Irazábal González
+//
+
+#if !defined(KRATOS_APPLY_FORCES_AND_MOMENTS_PROCESS )
+#define  KRATOS_APPLY_FORCES_AND_MOMENTS_PROCESS
+
+
+
+// System includes
+#include <string>
+#include <iostream>
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "includes/table.h"
+#include "includes/kratos_flags.h"
+#include "includes/kratos_parameters.h"
+#include "includes/model_part.h"
+#include "processes/process.h"
+#include "utilities/interval_utility.h"
+#include "utilities/python_function_callback_utility.h"
+#include "DEM_application_variables.h"
+
+namespace Kratos
+{
+  ///@addtogroup DEMApplication
+  ///@{
+
+  ///@name Kratos Globals
+  ///@{
+
+  ///@}
+  ///@name Type Definitions
+  ///@{
+
+  ///@}
+  ///@name  Enum's
+  ///@{
+
+  ///@}
+  ///@name  Functions
+  ///@{
+
+  ///@}
+  ///@name Kratos Classes
+  ///@{
+
+  /// Auxiliary process to apply forces and moments to particles.
+  /** This process sets the EXTERNAL_APPLIED_FORCE and EXTERNAL_APPLIED_MOMENT variables
+      over particles.
+   */
+  class KRATOS_API(DEM_APPLICATION) ApplyForcesAndMomentsProcess: public Process
+  {
+  public:
+      ///@name Type Definitions
+      ///@{
+
+      /// Pointer definition of ApplyForcesAndMomentsProcess
+      KRATOS_CLASS_POINTER_DEFINITION(ApplyForcesAndMomentsProcess);
+
+      ///@}
+      ///@name Life Cycle
+      ///@{
+
+      /// Constructor
+      ApplyForcesAndMomentsProcess(ModelPart& rModelPart, Parameters& rParameters);
```

Comment:
```suggestion
      ApplyForcesAndMomentsProcess(ModelPart& rModelPart, Parameters rParameters);
```

due to how the parameters handle the memory internal (basically with a shared ptr to the actual json object), `Parameters` should be passed by value